components/header-bar/src/online-status.js

```diff
@@ -0,0 +1,31 @@
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+import React from 'react'
+import styles from './online-status.styles.js'
+
+/** A badge to display online/offline status in the header bar */
+export function OnlineStatus({ status, info, mobile }) {
+    // Convert to title case
+    const displayStatus = status.replace(/(^|\s)\S/g, c => c.toUpperCase())
+
+    return (
+        <div
+            className={cx('container', mobile ? 'mobile' : 'desktop')}
+            data-test="headerbar-online-status"
+        >
+            {info && !mobile && <span className="info">{info}</span>}
+            <div className={cx('icon', status)}></div>
+            <span className="label">{displayStatus}</span>
+            {info && mobile && <span className="info-mobile">{info}</span>}
+            <style jsx>{styles}</style>
+        </div>
+    )
+}
+OnlineStatus.propTypes = {
+    /** Additional text to display beside badge */
+    info: PropTypes.string,
+    /** If true, uses mobile styles */
+    mobile: PropTypes.bool,
```

Comment:
Is "mobile" a term @dhis2/team-ux uses or should this be named e.g. `compact` instead to steer away from ending up with assumptions about what device type a user has? So we don't end up with a suite of props like: `mobile`, `desktop`, `tablet`, `ultra-wide`, etc.

Reply:
good point @varl – in documentation it's been referred to elsewhere as `small screen`, but I'm happy to establish `compact` as a catch-all term.

Reply:
If we predict other variants, we could use a string to indicate size: `screen="small"`, `screen="large", `screen="ultra-wide"`` ?

Reply:
Makes sense in this scenario, but I think in the future there might be reasons other than screen size to use a `compact` (or `small`) interface – like user preference, so I think any terms we set now should be applicable to other uses in the future.

I liked your suggestion of `compact` as a standalone label because it defines the outcome, rather than the cause. The `compact` outcome can have multiple causes (screen size, user preference, etc.). Practically, with a key:value pair that might look like `layout-density: compact`, `layout-density:  roomy`.

Reply:
Other UI components use a `dense` boolean prop - should we go with that for now, then refactor to an enum-type 'layout-density' prop when we make more designs for other densities?

Reply:
`OnlineStatus` is not exported, so it's not available "from the outside", so we can use `dense` for now and any refactor we do will be an implementation detail. That gives us some room to think about a convention for other responsive components too.